doc(acme) add missing Admin API section and explain `domains` behaviour (#4046)
Kong/docs.konghq.com: app/_hub/kong-inc/acme/_index.md

        The list of domains to create certificate for. To match subdomains under `example.com`, use `*.example.com`.
        Regex pattern is not supported. Note this config is only used to match domains, not to specify the Common Name
        or Subject Alternative Name to create certificates; each domain must have its own certificate.

>         ACME plugin checks this configuration before the presense of certificate in `storage` when serving certificate of a request.

Reviewer: ```suggestion
        The ACME plugin checks this configuration before checking any certificate in `storage` when serving the certificate of a request.
```

curl http://localhost:8001/acme -XPATCH
```

## Monitoring and debugging

The ACME plugin exposes several endpoints through Admin API that can be used for
debugging and monitoring certificate creation and renewal.

> - **POST /acme**: start applying or renewing certificate and return the result; available parameter:
>   - **host**: the domain to create certificate
>   - **test_http_challenge_flow**: when set, only check for configuration sanity.

Reviewer: I suggest using the formatting that we use for requests elsewhere (here's another plugin documented in the same way: https://docs.konghq.com/hub/kong-inc/proxy-cache-advanced/):

```suggestion
### Apply certificate

**Endpoint**

<div class="endpoint post">/acme</div>
Applies or renews the certificate and returns the result.

**Request body**

Attribute | Description
---:| ---
`host`<br>*required* | The domain where to create the certificate.
`test_http_challenge_flow`<br>*optional* | When set, only checks if the configuration is valid. Does not apply the certificate.
```

> - **PATCH /acme**: same as POST, but runs the process at background.

Reviewer: ```suggestion

### Update certificate

Apply or renew the certificate and return the result. Unlike `POST`, `PATCH` runs the process in the background. 

**Endpoint**

<div class="endpoint patch">/acme</div>

**Request body**

Attribute | Description
---:| ---
`host`<br>*required* | The domain where to create the certificate.
`test_http_challenge_flow`<br>*optional* | When set, only checks if the configuration is valid. Does not apply the certificate.
```

Am I correct in assuming that `PATCH` also takes the same arguments as `POST`? Also, can it apply a certificate if one doesn't exist, or does it update only?

> - **GET /acme/certificates**: list the certificate being created by ACME plugin; one can use this endpoint to monitor certificate existence and expiry.

Reviewer: ```suggestion

### Get ACME certificates

List the certificates being created by the ACME plugin. You can use this endpoint to monitor certificate existence and expiry.
 
**Endpoint**
<div class="endpoint get">/acme/certificates</div>
```

> - **GET /acme/certificates/:host**: list the certificate with specific host.

Reviewer: ```suggestion

### Get certificate by host

List certificates with a specific host.

**Endpoint**
<div class="endpoint get">/acme/certificates/{HOST}</div>

Attribute | Description
---:| ---
`HOST` | The IP or hostname of the host to target.
```

> Following is an example of the certificate listing API:

Reviewer: Example response for listing certificates:

> Following is an example of the certificate listing API:

Reviewer: ```suggestion
Example response for listing certificates:
```

> ```

Reviewer: ```suggestion
```json
```
